Plugins/Flow.Launcher.Plugin.Program/Programs/Win32.cs

```diff
@@ -545,5 +555,68 @@ public bool Equals([AllowNull] Win32 other)
 
             return UniqueIdentifier == other.UniqueIdentifier;
         }
+
+        private static IEnumerable<string> GetStartMenuPaths()
+        {
+            var directory1 = Environment.GetFolderPath(Environment.SpecialFolder.Programs);
+            var directory2 = Environment.GetFolderPath(Environment.SpecialFolder.CommonPrograms);
+            return new[]
+            {
+                directory1, directory2
+            };
+        }
+
+        public static void WatchProgramUpdate(Settings settings)
+        {
+            var paths = new List<string>();
+            if (settings.EnableStartMenuSource)
+                paths.AddRange(GetStartMenuPaths());
+
+            paths.AddRange(from source in settings.ProgramSources where source.Enabled select source.Location);
+
+            foreach (var directory in from path in paths where Directory.Exists(path) select path)
+            {
+                WatchDirectory(directory);
+            }
+
+            _ = Task.Run(MonitorDirectoryChangeAsync);
+        }
+
+        private static Channel<byte> indexQueue = Channel.CreateBounded<byte>(1);
+
+        public static async Task MonitorDirectoryChangeAsync()
+        {
+            var reader = indexQueue.Reader;
+            while (await reader.WaitToReadAsync())
+            {
+                await Task.Delay(500);
+                while (reader.TryRead(out _))
+                {
+                }
+                Main.IndexWin32Programs();
```

Comment:
why is this while loop empty?

Reply:
To pop all remaining reindex request out.